pkg/resource_builders/envoyconfig/listeners.go

```diff
@@ -0,0 +1,260 @@
+										m := map[string]*structpb.Value{
+											"authority":             structpb.NewStringValue("%REQ(:AUTHORITY)%"),
+											"bytes_received":        structpb.NewStringValue("%BYTES_RECEIVED%"),
+											"bytes_sent":            structpb.NewStringValue("%BYTES_SENT%"),
+											"duration":              structpb.NewStringValue("%DURATION%"),
+											"method":                structpb.NewStringValue("%REQ(:METHOD)%"),
+											"path":                  structpb.NewStringValue("%REQ(X-ENVOY-ORIGINAL-PATH?:PATH)%"),
+											"protocol":              structpb.NewStringValue("%PROTOCOL%"),
+											"response_code":         structpb.NewStringValue("%RESPONSE_CODE%"),
+											"response_code_details": structpb.NewStringValue("%RESPONSE_CODE_DETAILS%"),
+											"response_flags":        structpb.NewStringValue("%RESPONSE_FLAGS%"),
+											"listener":              structpb.NewStringValue(name),
+											"upstream_cluster":      structpb.NewStringValue("%UPSTREAM_CLUSTER%"),
+											"upstream_service_time": structpb.NewStringValue("%RESP(X-ENVOY-UPSTREAM-SERVICE-TIME)%"),
+											"user_agent":            structpb.NewStringValue("%REQ(USER-AGENT)%"),
+											"client_ip":             structpb.NewStringValue("%REQ(X-ENVOY-EXTERNAL-ADDRESS)%"),
+										}
+										if tls {
+											m["downstream_tls_cipher"] = structpb.NewStringValue("%DOWNSTREAM_TLS_CIPHER%")
+											m["downstream_tls_version"] = structpb.NewStringValue("%DOWNSTREAM_TLS_VERSION%")
+										}
+										return m
+									}(),
+								},
+							},
+						},
+					},
+				}
+
+				any, err := anypb.New(logfmt)
+				if err != nil {
+					panic(err)
+				}
+				return any
+			}(),
+		},
+	}}
+}
+
+func TransportSocket_v1(secretName string, http2 bool) *envoy_config_core_v3.TransportSocket {
+	return &envoy_config_core_v3.TransportSocket{
+		Name: "envoy.transport_sockets.tls",
+		ConfigType: &envoy_config_core_v3.TransportSocket_TypedConfig{
+			TypedConfig: func() *anypb.Any {
+				any, err := anypb.New(&envoy_extensions_transport_sockets_tls_v3.DownstreamTlsContext{
+					CommonTlsContext: &envoy_extensions_transport_sockets_tls_v3.CommonTlsContext{
+						TlsCertificateSdsSecretConfigs: []*envoy_extensions_transport_sockets_tls_v3.SdsSecretConfig{
+							{
+								Name: secretName,
+								SdsConfig: &envoy_config_core_v3.ConfigSource{
+									ConfigSourceSpecifier: &envoy_config_core_v3.ConfigSource_Ads{
+										Ads: &envoy_config_core_v3.AggregatedConfigSource{},
+									},
+									ResourceApiVersion: envoy_config_core_v3.ApiVersion_V3,
+								},
+							},
+						},
+						TlsParams: &envoy_extensions_transport_sockets_tls_v3.TlsParameters{
+							TlsMinimumProtocolVersion: envoy_extensions_transport_sockets_tls_v3.TlsParameters_TLSv1_2,
+						},
+						AlpnProtocols: func() []string {
+							if http2 {
+								return []string{"h2,http/1.1"}
+							} else {
+								return []string{"http/1.1"}
```

Comment:
Good question, I have no idea because sometimes when http1.1 is referred, http1.0 is also included. I'll have to test and see. I say that we merge like this and change it if we find it does not work. We could use the Raw feature and override the whole listener anyway if something does not work as expected.

Reply:
ok!